src/core/AutoRest.Core/Validation/Rule.cs

```diff
@@ -16,9 +16,15 @@ protected Rule()
         {
         }
 
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
```

Comment:
Yes indeed that's what we want but eventually. At this stage some of the rules are not mapped with any Ids ,hence i am leaving it virtual for now. Once we have assigned all the rules to Ids and ViolationCategory we'll make both abstract.